test/integration/tasks.test.js

```diff
@@ -1691,4 +1691,100 @@ describe("Tasks", function () {
       expect(res.body.message).to.be.equal(INTERNAL_SERVER_ERROR);
     });
   });
+
+  describe("GET /tasks/v1/:userId", function () {
+    it("Should return all the completed tasks of the user when query 'completed' is true and dev is true", async function () {
+      const { userId: assignedUser } = await userModel.addOrUpdate({
+        github_id: "prakashchoudhary07",
+        username: "user1",
+      });
+
+      const assignedTask = [
+        {
+          title: "Test task",
+          type: "feature",
+          endsOn: 1234,
+          startedOn: 4567,
+          status: "COMPLETED",
+          percentCompleted: 10,
+          participants: [],
+          assignee: "user1",
+          completionAward: { [DINERO]: 3, [NEELAM]: 300 },
+          lossRate: { [DINERO]: 1 },
+          isNoteworthy: true,
+        },
+        {
+          title: "Test task",
+          type: "feature",
+          endsOn: 1234,
+          startedOn: 4567,
+          status: "COMPLETED",
+          percentCompleted: 10,
+          participants: [],
+          assignee: "user1",
+          completionAward: { [DINERO]: 3, [NEELAM]: 300 },
+          lossRate: { [DINERO]: 1 },
+          isNoteworthy: true,
+        },
+      ];
+      await tasks.updateTask(assignedTask[0]);
+      await tasks.updateTask(assignedTask[1]);
+
+      const res = await chai
+        .request(app)
+        .get(`/tasks/v1/${assignedUser}/?dev=true&completed=true`)
+        .set("cookie", `${cookieName}=${authService.generateAuthToken({ userId: assignedUser })}`);
+
+      expect(res).to.have.status(200);
+      expect(res.body).to.be.a("array");
+      expect(res.body).to.satisfy((tasks) => tasks.every((task) => task.status === "COMPLETED"));
+    });
+
+    it("Should return all the tasks of the user when dev is true", async function () {
```

Comment:
we should write tests for accessing tasks of another user without proper authorization.